geo/Cargo.toml

```diff
@@ -30,7 +31,7 @@ proj = { version = "0.27.0", optional = true }
 robust = "1.1.0"
 rstar = "0.12.0"
 serde = { version = "1.0", optional = true, features = ["derive"] }
-i_overlay = "1.7.2"
+i_overlay = { version = "1.7.2", optional = true }
```

Comment:
IMO boolean operations are such a core feature that I wouldn't oppose `i_overlay` itself being required. But I think the multithreading should be optional.

Also, don't you want `default-features = false` here, so that the `i_overlay/allow_multithreading` feature here is only applied when `multithreading` is on?

Reply:
Ah yes great catch, I meant to add `default-features = false` here

Reply:
https://github.com/georust/geo/pull/1250/commits/6e2303904d3b3470e683f194a8e21cbe3786fa67

Reply:
>IMO boolean operations are such a core feature that I wouldn't oppose i_overlay itself being required. But I think the multithreading should be optional.

I agree boolean operations are important, but the success of `geo` up until the operations were added to the crate indicates many people don't need that functionality and the dependency tree that comes with it. So they should have the ability to opt-out. Adding the extra `cfg` check here is low maintenance cost for us

Reply:
Just to be clear, we've had *non optional* boolean operations in geo for years now. So if anyone is building geo with `no-default` features, this will be a breaking change for them. 

I personally have a slight preference for "make rayon optional but bool_ops mandatory" - because of the above breakage and adding configuration and documentation complexity for a feature on average outweighs the benefits of letting people fine tune in this case in my estimation.

But it's only a slight prefrence.

Question: if we break the no-default-features build, do we need to make another bump to `0.X.0`?

Reply:
> if anyone is building geo with no-default features, this will be a breaking change for them.

That hadn't occurred to me. I'm now also in the "slight preference for mandatory" camp.

Reply:
I still have a preference for making the bool ops stuff optional but it sounds like there's agreement about making it required so I'll just change it